src/GNDStk/gnds-convert-to-json.hpp

```diff
@@ -0,0 +1,134 @@
+
+/*
+Summary of the functions in this file:
+
+namespace detail {
+   1. prefix(unsigned long)
+
+   2. Node2json(gnds::Node, nlohmann::json, unsigned long)
+      ...uses (1)
+}
+
+3. convert(gnds::Tree, gnds::json)
+   ...uses (2)
+
+4. convert(gnds::xml,  gnds::json)
+   ...uses (3)
+
+5. convert(gnds::json, gnds::json)
+*/
+
+
+
+// -----------------------------------------------------------------------------
+// Helpers
+// -----------------------------------------------------------------------------
+
+namespace detail {
+
+// prefix
+inline std::string prefix(const unsigned long n)
+{
+   std::ostringstream oss;
+   oss << std::setfill('0') << std::setw(12) << n;
+   return oss.str();
+}
+
+
+
+// Node2json
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+bool Node2json(
```

Comment:
Does this part of the naming policy override the other part of the naming policy that says classes should be upper camel case? Here, "Node" refers to a class. So, unless I change the function's name to something entirely different (e.g. convert()), then either (1) I start with N and violate the function policy, or (2) I start with n and violate the class policy (at least indirectly, insofar as the "Node" part of the function's name refers to a class). A problem I've repeatedly encountered with naming conventions is that they almost always lead to conflicts like this (which is why I personally prefer to take the approach of the entire C++ language and library: just make everything lower-case - easy to remember and to type - and design names that are so clear and natural that nobody wonders if something is a class or a function because the answer is either obvious, or doesn't matter). When a naming-convention conflict does occur, we can easily decide which way to go, but users are left wondering which way we went. Unless they, you know, RTFM (Read The Manual), which almost nobody ever does. :-) In this case I may just change the name to something generic, like convert(). In fact, at the moment at least, you'll remember that I used an overloaded convert() for the various user-callable conversions between the major data types. For the lower-level functions like this (node, not whole tree), I'd intentionally given them more-specific names because doing so made them easier for me to find in the code, and ease of locating them seemed more important than name consistency because these are internal details, not part of the real API.

Reply:
There is no conflict here. This is a function so it should be `lowerCamelCase`.

Reply:
OK. I changed that a while back, actually, so it should be in the current code.